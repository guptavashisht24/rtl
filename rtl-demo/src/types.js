export const ADD_DATA = 'Add data';
export const REMOVE_DATA = 'Remove Data';