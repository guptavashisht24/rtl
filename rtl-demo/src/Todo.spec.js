import React from 'react';
import { render, cleanup } from '@testing-library/react';
import "jest-dom/extend-expect";
import ConnectedTodo, { Todo } from './todo';
import { exportAllDeclaration } from '@babel/types';



describe('Tests without connection', ()=>{
  afterEach(cleanup);

  it('matches snapshot',()=>{
    const { asFragment } = render(<Todo items = {[]}/>)
    expect(asFragment()).toMatchSnapshot()
  })

  it('has add button',()=>{
    const { getByText } = render(<Todo items = {[]}/>)
    getByText('ADD')
  })

  it('has remove button',()=>{
    const { getByText } = render(<Todo items = {[]}/>)
    getByText('REMOVE')
  })

  it('has remove button',()=>{
    const { getByText } = render(<Todo items = {[]}/>)
    getByText('REMOVE')
  })

  it('has list with 0 element',()=>{
    const { getByTestId } = render(<Todo items = {[]}/>)
    const todoList = getByTestId("todos-ul")
    expect(todoList.children.length).toBe(0)
  })
  
})


