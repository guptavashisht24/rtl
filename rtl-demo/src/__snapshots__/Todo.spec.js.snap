// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests without connection matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="flex"
    >
      <input
        type="text"
        value=""
      />
      <button
        type="button"
      >
        ADD
      </button>
      <button
        type="button"
      >
        REMOVE
      </button>
    </div>
    <ul />
  </div>
</DocumentFragment>
`;
